server/src/main/java/org/candlepin/model/EntitlementCurator.java

```diff
@@ -987,4 +875,175 @@ public Entitlement findUpstreamEntitlementForStack(Consumer consumer, String sta
 
         return (Entitlement) activeNowQuery.uniqueResult();
     }
+
+    /**
+     * Finds the dependent entitlements for the specified entitlements, matching
+     * on provided products only. Dependent entitlements are those who's
+     * content are being modified by the consumption of another entitlement.
+     *
+     * @param entitlementIds the entitlements to match on.
+     * @return the set of entitlement IDs for the matched modifier entitlements.
+     */
+    private Set<String> findDependentEntitlementsByProvidedProduct(Iterable<String> entitlementIds) {
+        String queryStr = "SELECT DISTINCT e2.id " +
+            // Required entitlement
+            "FROM cp_entitlement e1 " +
+            // Required entitlement => required pool
+            "JOIN cp2_pool_provided_products ppp1 ON ppp1.pool_id = e1.pool_id " +
+            // Required pool => required product
+            "JOIN cp2_products p ON p.uuid = ppp1.product_uuid " +
+            // Required product => conditional content
+            "JOIN cp2_content_modified_products cmp ON cmp.element = p.product_id " +
+            // Conditional content => dependent product
+            "JOIN cp2_product_content pc ON pc.content_uuid = cmp.content_uuid " +
+            // Dependent product => dependent pool
+            "JOIN cp2_pool_provided_products ppp2 ON ppp2.product_uuid = pc.product_uuid " +
+            // Dependent pool => dependent entitlement
+            "JOIN cp_entitlement e2 ON e2.pool_id = ppp2.pool_id " +
+            "WHERE e1.consumer_id = e2.consumer_id " +
+            "  AND e1.id != e2.id " +
+            "  AND e2.dirty = false " +
+            "  AND e1.id IN (:entitlement_ids)" +
+            "  AND e2.id NOT IN (:entitlement_ids)";
+
+        Query query = getEntityManager().createNativeQuery(queryStr);
+
+        Set<String> result = new HashSet<String>();
+        if (entitlementIds == null || !entitlementIds.iterator().hasNext()) {
+            return result;
+        }
+
+        int blockSize = Math.min(this.getInBlockSize(), this.getQueryParameterLimit());
```

Comment:
This query, however, *does* need the division by two, as we have two blocks of params in the query.